c10/core/DispatchKey.h

```diff
@@ -360,54 +461,67 @@ enum class DispatchKey : uint8_t {
 // built-in autograd formulas for operators are not appropriate.
 
 static_assert(
-    static_cast<uint8_t>(DispatchKey::NumDispatchKeys) < 64,
+    (static_cast<uint8_t>(BackendBit::EndOfBackendKeys) +
+    static_cast<uint8_t>(DispatchKey::EndOfFunctionalityKeys)) <= 64,
     "DispatchKey is used as index into 64-bit bitmask; you must have less than 64 entries");
 
-#if defined(C10_MOBILE_TRIM_DISPATCH_KEYS)
-/**
- * The method below maps the dispatch key in the enum DispatchKey to an
- * integer index in the dispatchTable_ array in OperatorEntry. The array
- * is trimmed for mobile to reduce peak memory usage since it's
- * unnecessary to reserve additional space for dispatch keys that will
- * never be used on mobile.
- */
-C10_API constexpr int getDispatchTableIndexForDispatchKey(DispatchKey dk) {
-  switch (dk) {
-    case DispatchKey::Undefined:
-      return 0;
-    case DispatchKey::CPU:
-      return 1;
-    case DispatchKey::QuantizedCPU:
-      return 2;
-    case DispatchKey::SparseCPU:
-      return 3;
-    case DispatchKey::BackendSelect:
-      return 4;
-    case DispatchKey::ADInplaceOrView:
-      return 5;
-    case DispatchKey::AutogradOther:
-      return 6;
-    case DispatchKey::AutogradCPU:
-      return 7;
-    case DispatchKey::NumDispatchKeys: // Sentinel, end of runtime keys.
-      return 8;
-    default:
-      return -1;
+// Check if a DispatchKey is an alias mapping to other runtime keys.
+constexpr bool isAliasDispatchKey(DispatchKey k) {
+  return k >= DispatchKey::StartOfAliasKeys && k <= DispatchKey::EndOfAliasKeys;
+}
+
+// Check if a DispatchKey is a per-backend functionality key
+// [Note: Per-Backend Functionality Dispatch Keys]
+// TODO: fill this in.
+constexpr bool isPerBackendFunctionalityKey(DispatchKey k) {
+  if (k == DispatchKey::Dense ||
+      k == DispatchKey::Quantized ||
+      k == DispatchKey::Sparse ||
+      k == DispatchKey::AutogradFunctionality) {
```

Comment:
One fun thing about DispatchKeySet no longer being a true "set" is that I have to think about if `DispatchKeySet({DispatchKey::Dense, DispatchKey::Quantized, DispatchKey::Sparse, DispatchKey::AutogradFunctionality}).has(k)` works or not...

Reply:
Yep. I do it by keeping the "building block vs. runtime key" distinction clear in my head. All of the keys in that example above are building block keys, which means that `.has(k)` won't return true for any runtime keys.